test/libdnf5/module/test_module.cpp

```diff
@@ -653,3 +655,112 @@ void ModuleTest::test_module_enable() {
         }
     }
 }
+
+
+void ModuleTest::test_module_disable() {
+    add_repo_repomd("repomd-modules");
+
+    // Add module disable goal operation
+    libdnf5::Goal goal(base);
+    goal.add_module_disable("fruit-salad:main");
+    auto transaction = goal.resolve();
+
+    // Active modules contain the the default stream of module berries and its dependency gooseberry:5.5
+    std::vector<std::string> expected_active_module_specs{
+        "berries:main:4:6c81f848:x86_64", "gooseberry:5.5:2:72aaf46b6:x86_64", "gooseberry:5.5:3:72aaf46b6:x86_64"};
+    std::vector<std::string> active_module_specs;
+    for (auto & module_item : base.get_module_sack()->get_active_modules()) {
+        active_module_specs.push_back(module_item->get_full_identifier());
+    }
+    std::sort(active_module_specs.begin(), active_module_specs.end());
+    CPPUNIT_ASSERT_EQUAL(expected_active_module_specs, active_module_specs);
+
+    // Run the transaction
+    CPPUNIT_ASSERT_EQUAL(libdnf5::base::Transaction::TransactionRunResult::SUCCESS, transaction.run());
+
+    auto system_state = (base.*get(priv_impl()))->get_system_state();
+
+    // Module fruit-salat is DISABLED because it was explicitly disabled
+    CPPUNIT_ASSERT_EQUAL(
+        libdnf5::system::ModuleState({"", ModuleStatus::DISABLED, {}}), system_state.get_module_state("fruit-salad"));
+
+    // None of the other modules is DISABLED
+    for (auto [name, module_state] : system_state.get_module_states()) {
+        if (name != "fruit-salad") {
+            CPPUNIT_ASSERT_EQUAL(libdnf5::system::ModuleState({"", ModuleStatus::AVAILABLE, {}}), module_state);
+        }
+    }
+}
+
+
+void ModuleTest::test_module_disable_enabled() {
+    add_repo_repomd("repomd-modules");
+
+    // Set state of module berries to ENABLED
+    (base.*get(priv_impl()))
+        ->get_system_state()
+        .set_module_state("berries", libdnf5::system::ModuleState({"main", ModuleStatus::ENABLED, {}}));
```

Comment:
I am accessing the system state here and changing the status of one module. You may notice the system state is accessed later as well. But for some reason, if I use the same instance for the system state, the state change doesn't propagate correctly. I will investigate this further, but could it be related to the private accessor hack?

Reply:
I am sorry, I have no idea

Reply:
I can only guess that if the modification are not propagated, it sounds like that changes are performed on copy instance.